cypress/integration/queryBuilder.spec.js

```diff
@@ -0,0 +1,32 @@
+describe('test query builder', () => {
+  beforeEach(() => {
+    cy.server();
+
+    cy.route('GET', 'api/repository/v1/datasets/**').as('getDataset');
+    cy.route('GET', 'api/repository/v1/datasets/**/policies').as('getDatasetPolicies');
+
+    cy.visit('/login/e2e');
+    cy.get('#tokenInput').type(Cypress.env('GOOGLE_TOKEN'), {
+      log: false,
+      delay: 0,
+    });
+    cy.get('#e2eLoginButton').click();
+
+    cy.contains('V2F_GWAS_Summary_Stats').should('be.visible');
+    cy.contains('V2F_GWAS_Summary_Stats').click();
+    cy.wait(['@getDataset', '@getDatasetPolicies']);
+    cy.get('[data-cy=queryDatasetButton]').click();
+  });
+
+  it('does render', () => {});
+
+  it('applies filters', () => {
+    cy.get('div.MuiButtonBase-root:nth-child(2) > svg:nth-child(1)').click();
```

Comment:
For selectors like this that aren't semantic like the ones below, it'd be useful to add a comment above about what is actually getting selected.